### PR TITLE
Adding new S2 stuff

### PR DESCRIPTION
**Pull request by tasodorff**

### CODE REVIEW

**Comment by charbull** [comment]
@tasodorff please remove the bin files *.pyc

File: ontology/yaml/resources/HVAC/entity_types/GENERALTYPES.yaml [hunk 1]
```diff
@@ -251,3 +251,36 @@ VLV:
   implements:
   - EQUIPMENT
 
+RSR:
+  description: "A distribution riser."
+  is_abstract: true
+
+RARSR:
+  description: "A distribution riser."
```
Comment (berkoben) [inline ontology/yaml/resources/HVAC/entity_types/GENERALTYPES.yaml:259]: what kind of rider?  (here and below)

File: ontology/yaml/resources/HVAC/entity_types/RSR.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+
+
+CHWRSR_UNDEFINED:
+  description: "Chilled water distribution riser without any data points."
+  implements:
+  - CHWRSR
+
+HWSRSR_UNDEFINED:
+  description: "Chilled water distribution riser without any data points."
```
Comment (berkoben) [inline ontology/yaml/resources/HVAC/entity_types/RSR.yaml:29]: hot

File: ontology/yaml/resources/HVAC/entity_types/RSR.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  implements:
+  - HWSRSR
+
+FARSR_UNDEFINED:
+  description: "Fresh air distribution riser without any data points."
+  implements:
+  - FARSR
+
+RARSR_UNDEFINED:
+  description: "Fresh air distribution riser without any data points."
```
Comment (berkoben) [inline ontology/yaml/resources/HVAC/entity_types/RSR.yaml:39]: return?

File: ontology/yaml/resources/HVAC/entity_types/AHU.yaml [hunk 4]
```diff
@@ -2230,6 +2230,11 @@ AHU_BSPC_CHWSC_ECON_EFSS_EFVSC_HWSC_PHWSC_SFSS_SFVSC_SSPC:
   - SFVSC
   - SSPC
 
+AHU_UNDEFINED:
+  description: "An AHU with no telemetry data."
```
Comment (berkoben) [inline ontology/yaml/resources/HVAC/entity_types/AHU.yaml:2234]: Note in description that this should only be used in special cases where AHU information is not available.

File: ontology/yaml/resources/HVAC/entity_types/VAV.yaml [hunk 5]
```diff
@@ -348,6 +348,11 @@ VAV_ED_RMM:
   - ED
   - RMM
 
+VAV_UNDEFINED:
```
Comment (berkoben) [inline ontology/yaml/resources/HVAC/entity_types/VAV.yaml:351]: Note in description that this should only be used in special cases where VAV information is not available.

File: tools/validators/instance_validator/validate/handler.py [hunk 6]
```diff
@@ -97,7 +98,7 @@ def ValidateEntities(self, entity_instances):
     for entity_name, current_entity_instance in entity_instances.items():
       if not current_entity_instance.IsValidEntityInstance(entity_instances):
         print(entity_name, 'is not a valid instance')
-        sys.exit(0)
+        #sys.exit(0)
```
Comment (berkoben) [inline tools/validators/instance_validator/validate/handler.py:101]: revert.

File: tools/validators/instance_validator/validate/handler.py [hunk 7]
```diff
@@ -54,6 +54,7 @@ def deserialize(yaml_file, universe):
   return entity_instances, parsed
 
 
+
```
Comment (berkoben) [inline tools/validators/instance_validator/validate/handler.py:57]: revert

**Review by berkoben — CHANGES_REQUESTED** [review]

File: ontology/yaml/resources/HVAC/entity_types/GENERALTYPES.yaml (hunk 1 above)
Comment (tasodorff) [reply ontology/yaml/resources/HVAC/entity_types/GENERALTYPES.yaml:259]: fixed

File: ontology/yaml/resources/HVAC/entity_types/RSR.yaml (hunk 2 above)
Comment (tasodorff) [reply ontology/yaml/resources/HVAC/entity_types/RSR.yaml:29]: fixed

File: ontology/yaml/resources/HVAC/entity_types/AHU.yaml (hunk 4 above)
Comment (tasodorff) [reply ontology/yaml/resources/HVAC/entity_types/AHU.yaml:2234]: added

File: ontology/yaml/resources/HVAC/entity_types/VAV.yaml (hunk 5 above)
Comment (tasodorff) [reply ontology/yaml/resources/HVAC/entity_types/VAV.yaml:351]: added

File: tools/validators/instance_validator/validate/handler.py [hunk 8]
```diff
@@ -97,7 +98,7 @@ def ValidateEntities(self, entity_instances):
     for entity_name, current_entity_instance in entity_instances.items():
```
Comment (berkoben) [inline tools/validators/instance_validator/validate/handler.py:98]: add above for loop:
has_invalid_instances = False

File: tools/validators/instance_validator/validate/handler.py (hunk 6 above)
Comment (berkoben) [reply tools/validators/instance_validator/validate/handler.py:101]: replace this with:
 has_invalid_instances = True

File: tools/validators/instance_validator/validate/handler.py [hunk 9]
```diff
@@ -97,7 +98,7 @@ def ValidateEntities(self, entity_instances):
...
       if not current_entity_instance.IsValidEntityInstance(entity_instances):
         print(entity_name, 'is not a valid instance')
-        sys.exit(0)
+        #sys.exit(0)
       if current_entity_instance.type_name.lower() == 'building':
         building_found = True
```
Comment (berkoben) [inline tools/validators/instance_validator/validate/handler.py:104]: add (outside loop):
if has_invalid_instances:
  sys.exit(0)

**Review by berkoben — CHANGES_REQUESTED** [review]

**Comment by charbull** [comment]
@tasodorff please remove this file : )

send it internally please
tools/validators/instance_validator/s2new.yaml

File: ontology/yaml/resources/HVAC/entity_types/AHU.yaml [hunk 10]
```diff
@@ -2230,6 +2230,11 @@ AHU_BSPC_CHWSC_ECON_EFSS_EFVSC_HWSC_PHWSC_SFSS_SFVSC_SSPC:
   - SFVSC
   - SSPC
 
+AHU_UNDEFINED:
+  description: "An AHU with no telemetry data. Note this is non-canonical, as it is typically expected that telemetry will normally be available."
```
Comment (berkoben) [inline ontology/yaml/resources/HVAC/entity_types/AHU.yaml:2234]: -normally

File: ontology/yaml/resources/HVAC/entity_types/GENERALTYPES.yaml [hunk 11]
```diff
@@ -251,3 +251,36 @@ VLV:
...
+  is_abstract: true
+  implements:
+  - RSR
+
+CHWRSR:
+  description: "A chilled water distribution riser."
+  is_abstract: true
+  implements:
+  - RSR
+  - CHWS
```
Comment (berkoben) [inline ontology/yaml/resources/HVAC/entity_types/GENERALTYPES.yaml:275]: Implementing CHWS seems incorrect.  I think what you mean is that it should be part of a CHWS (which we cannot represent until we implement connection constraints)

File: ontology/yaml/resources/HVAC/entity_types/VAV.yaml [hunk 12]
```diff
@@ -348,6 +348,11 @@ VAV_ED_RMM:
...
+  description: "VAV with no defined fields. Note this is non-canonical, as it is typically expected that telemetry will normally be available."
```
Comment (berkoben) [inline ontology/yaml/resources/HVAC/entity_types/VAV.yaml:352]: -normally (here and other similar descriptions)

**Review by berkoben — CHANGES_REQUESTED** [review]

**Review by berkoben — APPROVED** [review]

**Review by charbull — APPROVED** [review]

**Pull request merged**